### PR TITLE
NCL-4762 Add dependency analyser and correct dependencies. Upgrade to…

### PR DESCRIPTION
**Pull request by rnc**
… PME 3.8

### CODE REVIEW

**Comment by rnc** [comment]
Pre-requisite for #132

**Comment by rnc** [comment]
( Note - this picked up we were using Maven 3.0.3 instead of 3.5.0 )

File: analyzer/build.gradle.kts [hunk 1]
```diff
@@ -29,18 +29,37 @@ dependencies {
...
-    testCompile("org.jboss.byteman:byteman-install:${project.extra.get("bytemanVersion")}")
     testCompile (files ("${System.getProperty("java.home")}/../lib/tools.jar") )
-    testCompile("org.assertj:assertj-core:3.12.2")
     testCompile("org.mockito:mockito-core:2.27.0")
     testCompile("com.github.tomakehurst:wiremock-jre8:2.23.2")
+
+    // Ignore our own modules; avoid https://github.com/wfhartford/gradle-dependency-analyze/issues/83
+    permitUsedUndeclared(project(":common"))
+    // Groovy is built into Gradle
+    permitUsedUndeclared("org.codehaus.groovy:groovy:2.5.7")
```
Comment (geoand) [inline analyzer/build.gradle.kts]: Not sure I understand this one, why is it needed?

Comment (rnc) [reply analyzer/build.gradle.kts]: Do you mean the `permitUsedUndeclared` /  `permitUsedUndeclared` or specifically for groovy ?

Comment (geoand) [reply analyzer/build.gradle.kts]: Specifically for groovy.

Comment (rnc) [reply analyzer/build.gradle.kts]: It reports groovy as `permitUsedUndeclared` but groovy is supplied by gradle so we don't need to declare it

Comment (geoand) [reply analyzer/build.gradle.kts]: Interesting stuff, I had never seen that before

Comment (rnc) [reply analyzer/build.gradle.kts]: It's from this plugin https://github.com/wfhartford/gradle-dependency-analyze

Comment (geoand) [reply analyzer/build.gradle.kts]: :+1:

**Pull request merged**